src/yunohost/app.py

```diff
@@ -81,37 +82,69 @@ def app_listlists():
 
 def app_fetchlist(url=None, name=None):
     """
-    Fetch application list from app server
+    Fetch application list(s) from app server
 
     Keyword argument:
         name -- Name of the list (default yunohost)
         url -- URL of remote JSON list (default https://app.yunohost.org/official.json)
 
     """
     # Create app path if not exists
-    try: os.listdir(repo_path)
-    except OSError: os.makedirs(repo_path)
+    if not os.path.exists(repo_path):
+        os.makedirs(repo_path)
 
-    if url is None:
-        url = 'https://app.yunohost.org/official.json'
-        name = 'yunohost'
+    if url is not None and name:
+        app_lists = [(url, name)]
+    elif url is not None and name is None:
+        raise MoulinetteError(errno.EINVAL,
+                              m18n.n('custom_appslist_name_required'))
     else:
-        if name is None:
-            raise MoulinetteError(errno.EINVAL,
-                                  m18n.n('custom_appslist_name_required'))
+        app_lists = []
+        for filename in [x for x in os.listdir(repo_path) if x.endswith(".json")]:
+            base_filename = filename[:-len(".json")]
+            url_filename_path = os.path.join(repo_path, base_filename + ".url")
+
+            if os.path.exists(url_filename_path):
+                url = open(url_filename_path, "r").read().strip()
+            else:  # XXX backward compatible code, YunoHost never store the url of the list else where
+                cron_file_path = "/etc/cron.d/yunohost-applist-%s" % base_filename
+
+                if not os.path.exists(cron_file_path):
+                    logger.warning("neither %s nor %s exist, is the app list '%s' correctly installed on the system?" %\
```

Comment:
This will display this English message to the user. You should add a new translation key with this message and make use of it with `m18n.n(...)`.